### PR TITLE
Separate responseOrError field to be 2 different fields

### PR DESCRIPTION
**Pull request by jonasgraterol**
Separate `responseOrError` into 2 different fields to have a more clear logger in **goodeggs-server** see [this comment](https://www.pivotaltracker.com/story/show/172844548/comments/229796750) for details

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -89,7 +90,7 @@ async function retryFetch(
         const res = await fetch(requestUrl, requestOptions);
         if (shouldRetry(res)) throwRetryError(null);
         jsonFetchOptions.onRequestEnd?.({
-          responseOrError: res,
+          status: res,
```
Comment (serhalp) [inline src/index.ts:93]: We just want the status. That's the main problem we're trying to solve with this PR: the response is too huge to be logged!
```suggestion
          status: res.status,
```

**Review by serhalp — CHANGES_REQUESTED** [review]
Thanks!

File: README.md [hunk 2]
```diff
@@ -128,7 +128,7 @@ If given, `onRequestEnd` is called with:
   // ... all the original json-fetch options, plus:
   url: string;
   retryCount: number;
-  responseOrError: Response | Error;
+  status: Response['status'];
```
Comment (serhalp) [inline README.md:131]: Missing docs for the `error` option here
```suggestion
  status?: Response['status'];
  error?: Error,
```

**Review by serhalp — APPROVED** [review]
👍🏼

File: src/index.ts [hunk 3]
```diff
@@ -89,7 +90,7 @@ async function retryFetch(
         const res = await fetch(requestUrl, requestOptions);
         if (shouldRetry(res)) throwRetryError(null);
         jsonFetchOptions.onRequestEnd?.({
-          responseOrError: res,
+          status: res.status,
```
[the remark this answers is not in the document]
Comment (jonasgraterol) [reply src/index.ts:93]: Makes sense, I'll add more coverage

**Review by serhalp — APPROVED** [review]
👍🏼

**Comment by jonasgraterol** [comment]
Thanks for your review

**Pull request merged**